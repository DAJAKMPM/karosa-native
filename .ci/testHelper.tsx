import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
