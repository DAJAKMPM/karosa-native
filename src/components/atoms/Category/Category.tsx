/**
 *
 * Category
 * @format
 *
 */

import React, { FC } from "react";
import { View, Pressable } from "react-native";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import CategoryStyles from "./styles";

const Category: FC<PropsType> = (props) => {
  const { name, onPress } = props;

  return (
    <Pressable onPress={onPress}>
      <View style={CategoryStyles.categoryContainer}>
        <View style={CategoryStyles.categoryCircle} />
        <View style={CategoryStyles.nameContainer}>
          <Text text={name} textStyle={CategoryStyles.txtName} />
        </View>
      </View>
    </Pressable>
  );
};

export default Category;
