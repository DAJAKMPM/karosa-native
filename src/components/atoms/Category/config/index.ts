/**
 *
 * @format
 *
 */

export const ICON_SIZE = 65;
