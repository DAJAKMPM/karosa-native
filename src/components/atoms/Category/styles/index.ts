/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const CategoryStyles = StyleSheet.create({
  categoryContainer: {
    alignItems: "center",
    width: 100,
    marginBottom: 10,
  },
  categoryCircle: {
    height: 72,
    width: 72,
    backgroundColor: theme.colors.dark10,
    borderRadius: 100,
  },
  nameContainer: {
    marginTop: 4,
    width: 50,
  },
  txtName: {
    ...theme.textLight,
    textAlign: "center",
    fontWeight: "300",
  },
});

export default CategoryStyles;
