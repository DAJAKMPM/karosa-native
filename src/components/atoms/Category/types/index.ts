/**
 *
 * @format
 *
 */

export type PropsType = {
  id: number;
  name: string;
  iconName: string;
  onPress?(): void;
};
