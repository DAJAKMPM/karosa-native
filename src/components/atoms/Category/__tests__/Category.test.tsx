import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import { noop } from "lodash";
import Category from "../Category";

describe("Category Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Category
        id={2}
        iconName="preOrder"
        name="categoryVegetables"
        onPress={noop}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
