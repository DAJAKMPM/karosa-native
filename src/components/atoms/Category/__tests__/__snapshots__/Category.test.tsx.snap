// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 10,
        }
      }
    >
      <View>
        <SvgMock
          height={65}
          width={65}
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 4,
            "width": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1C1C1C",
              "fontFamily": "SFProText-Regular",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 14.06,
              "textAlign": "center",
            }
          }
        >
          categoryVegetables
        </Text>
      </View>
    </View>
  </View>
</View>
`;
